### DISCUSSION TITLE
DatePickerTextBox ignores custom StringFormat

### DISCUSSION THREAD

**Discussion by bergziege** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2952] [Q&A]
Hi all,

having a bit of a special customer request who wants the UI run in `en` locale but the date formatted like `dd.MM.yyyy`.
Before using MD I used a custom style for the date picker text box. But that is ignored by MD.

So is there any way to change the display format of the picker to be `dd.MM.yyyy` and keep the rest as `en`?
Or is there any `en` locale using the `.` format for dates?

Cheers

---------------------

Here is what worked before MD:

```
<Style TargetType="{x:Type DatePickerTextBox}" BasedOn="{StaticResource {x:Type DatePickerTextBox}}">
        <Setter Property="Template">
            <Setter.Value>
                <ControlTemplate>
                    <TextBox x:Name="PART_TextBox"
                             Text="{Binding Path=SelectedDate, StringFormat='dd.MM.yyyy', RelativeSource={RelativeSource AncestorType={x:Type DatePicker}}}">
                    </TextBox>
                </ControlTemplate>
            </Setter.Value>
        </Setter>
    </Style>
```

**Comment by Keboo** [comment]
The reason for this is because the DatePicker uses a `StaticResource` for the `DatePickerTextBox` (see https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/master/MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.DatePicker.xaml#L504). 

At present there is not a great option for doing this. The best I could suggest would be to copy in the `DatePicker` template into your project, and update the style/template for `DatePickerTextBox`
